serverlessworkflow/modules/ROOT/pages/eventing/working-with-callbacks.adoc

```diff
@@ -1 +1,108 @@
+----
+"events": [
+    {
+      "name": "resumeEvent",
+      "source": "",
+      "type": "resume"
+    },
+    {
+      "name": "waitEvent",
+      "source": "",
+      "type": "wait"
+    }
+  ]
+----
+
+Then it declares a Callback state that will publish an event of type `resume` and wait for an event of type `wait`. The Cloud Event being published contains a `move` data field. The event being received is expected to have a `result` data field, which according to the link:{spec_doc_url}#event-data-filters[eventDataFilter], will be added to the workflow model as `move` field. 
+
+[code,json]
+----
+{
+      "name": "waitForEvent",
+      "type": "callback",
+      "action": {
+        "name": "publishAction",
+        "eventRef": {
+          "triggerEventRef": "resumeEvent",
+          "data": "{move: \"This is the initial data in the model\"}"
+        }
+      },
+      "eventRef": "waitEvent",
+      "eventDataFilter": {
+        "data": ".result",
+        "toStateData": ".move"
+      },
+      "transition": "finish"
+    }
+----
+
+The event of type `resume` is consumed by a link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus/src/main/java/org/kie/kogito/examples/PrintService.java[Java listener]. This Java listener publishes a new event of type `wait`
+
+[code,java]
+----
+
+    private String generateCloudEvent(String id, String input) {
+        Map<String, Object> eventBody = new HashMap<>();
+        eventBody.put("result", input + " and has been modified by the event publisher");
+        eventBody.put("dummyEventVariable", "This will be discarded by the process");
+        try {
+            return objectMapper.writeValueAsString(CloudEventBuilder.v1()
+                    .withId(UUID.randomUUID().toString())
+                    .withSource(URI.create(""))
+                    .withType("wait")
+                    .withTime(OffsetDateTime.now())
+                    .withExtension(CloudEventExtensionConstants.PROCESS_REFERENCE_ID, id)
+                    .withData(objectMapper.writeValueAsBytes(eventBody))
+                    .build());
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(e);
+        }
+
+    }
+----
+
+The cloud event published by the listener modifies the original value of the `move` field and publish it as the `result` field. Moreover, it contains a header attribute, named `kogitoprocrefid`, holding the process instance id of the workflow that published the event. As mentioned before, this header is pivotal, because, when not using correlation, it is the only way for the callback state to know that this event should be used to resume the flow. 
+
+Remember that every workflow has a unique instance id that is automatically included in any event published by Kogito. As you can see in the code snippet below, the event listener takes the process instance id from a cloud event attribute named `kogitoprocinstanceid` of the event being consumed. 
```

Comment:
WDYT about adding a reference here or just mentioning from where this snipped was extracted? Maybe just a link to the actual example file like you did before.

Reply:
I renamed the titles, since this excerpt has the same origin than the previous one